### PR TITLE
Fix wrong positioning on first click events

### LINKED ISSUE #94: Contextmenu has wrong position
... on first and second click  in right and/or lower sector of the map

### PR DESCRIPTION
**Pull request by taulinger**
Re-implementation of the positioning mechanism of the `MapContextMenu`. Fixes #94 .

### CODE REVIEW

File: src/modules/map/components/contextMenu/MapContextMenu.js [hunk 1]
```diff
@@ -22,77 +21,66 @@ export class MapContextMenu extends BaElement {
...
-		const windowHeight = window.innerHeight - (window.innerHeight * offsetBorderInPercent);
-
-		if ((windowWidth - coordinate[0]) < menuWidth) {
-			values.vAlignment = 'right';
-			values.left = coordinate[0] - menuWidth + 'px';
-		}
-		else {
-			values.vAlignment = 'left';
-			values.left = coordinate[0] + 'px';
+		//windiw sector the click event occurred:
```
Comment (thiloSchlemmer) [inline src/modules/map/components/contextMenu/MapContextMenu.js:29]: Typo

File: src/modules/map/components/contextMenu/MapContextMenu.js [hunk 2]
```diff
@@ -22,77 +21,66 @@ export class MapContextMenu extends BaElement {
...
+			return 1;
 		}
-		else {
-			values.hAlignment = 'top';
-			values.top = coordinate[1] + vOffset + 'px';
+		else if (coordinate[0] > widthBorder && coordinate[1] > heightBorder) {
+			return 2;
 		}
-		return values;
+		return 3;
```
Comment (thiloSchlemmer) [inline src/modules/map/components/contextMenu/MapContextMenu.js:42]: Do we discussed  already named_values for magic numbers like 0-3? For instance `const VIEW_QUADRANT_UPPER_LEFT =0` etc.

**Review by thiloSchlemmer — APPROVED** [review]

File: src/modules/map/components/contextMenu/MapContextMenu.js (hunk 1 above)
Comment (taulinger) [reply src/modules/map/components/contextMenu/MapContextMenu.js:29]: thx

File: src/modules/map/components/contextMenu/MapContextMenu.js (hunk 2 above)
Comment (taulinger) [reply src/modules/map/components/contextMenu/MapContextMenu.js:42]: Good point, but let's wait until we can use static fields. We want to use such named values also outside this single method. From my point of view it's confusing, if we would use member variables for that case, they should represent state...

**Pull request merged**